operator/internal/validation/openshift/recordingrule.go

```diff
@@ -14,6 +14,15 @@ import (
 func RecordingRuleValidator(_ context.Context, recordingRule *lokiv1.RecordingRule) field.ErrorList {
 	var allErrs field.ErrorList
 
+	validateTenantIDs, fieldErr := tenantIDValidationEnabled(recordingRule.Annotations)
+	if fieldErr != nil {
+		return field.ErrorList{fieldErr}
+	}
+
+	if !validateTenantIDs {
+		return allErrs
```

Comment:
Would it be better to move the declaration of `allErrs` below this `if` clause and just `return nil` here?

```suggestion
		return nil
```